agent/pkg/tunnel/config/config.go

```diff
@@ -1,31 +1,40 @@
 package config
 
-type Tunnel struct {
-	// Enable indicates whether EdgeHub is enabled,
-	// if set to false (for debugging etc.), skip checking other EdgeHub configs.
+import (
+	"os"
+
+	"github.com/kubeedge/edgemesh/common/acl"
+	meshConstants "github.com/kubeedge/edgemesh/common/constants"
+	"k8s.io/klog/v2"
+)
+
+type TunnelAgentConfig struct {
+	// Enable indicates whether TunnelAgent is enabled,
+	// if set to false (for debugging etc.), skip checking other TunnelAgent configs.
 	// default true
 	Enable bool `json:"enable"`
-	// Heartbeat indicates heart beat (second)
-	// default 15
-	Heartbeat int32 `json:"heartbeat,omitempty"`
-	// TLSCAFile set ca file path
-	// default "/etc/kubeedge/ca/rootCA.crt"
-	TLSCAFile string `json:"tlsCaFile,omitempty"`
-	// TLSCertFile indicates the file containing x509 Certificate for HTTPS
-	// default "/etc/kubeedge/certs/server.crt"
-	TLSCertFile string `json:"tlsCertFile,omitempty"`
-	// TLSPrivateKeyFile indicates the file containing x509 private key matching tlsCertFile
-	// default "/etc/kubeedge/certs/server.key"
-	TLSPrivateKeyFile string `json:"tlsPrivateKeyFile,omitempty"`
-	// Token indicates the priority of joining the cluster for the edge
-	Token string `json:"token"`
-	// HTTPServer indicates the server for edge to apply for the certificate.
-	HTTPServer string `json:"httpServer,omitempty"`
-	// RotateCertificates indicates whether edge certificate can be rotated
-	// default true
-	RotateCertificates bool `json:"rotateCertificates,omitempty"`
-	// HostnameOverride indicates hostname
-	// default os.Hostname()
-	HostnameOverride string `json:"hostnameOverride,omitempty"`
+	// TunnelACLConfig indicates the set of tunnel agent config about acl
+	acl.TunnelACLConfig
+	// NodeName indicates the node name of tunnel agent
+	NodeName string `json:"nodeName"`
+	// ListenPort indicates the listen port of tunnel agent
+	// default 10006
+	ListenPort int `json:"listenPort"`
+}
+
+func NewTunnelAgentConfig() *TunnelAgentConfig {
+	nodeName, isExist := os.LookupEnv(meshConstants.MY_NODE_NAME)
+	if !isExist {
+		klog.Fatalf("env %s not exist", meshConstants.MY_NODE_NAME)
+		os.Exit(1)
```

Comment:
`klog.Fatalf` will exit the program immediately.

Reply:
yes, os.Exit(1) should't appear on this area